src/server/routes/register.ts

```diff
@@ -111,10 +112,13 @@ const OktaRegistration = async (
   req: Request,
   res: ResponseWithRequestState,
 ) => {
-  const { email = '' } = req.body;
-  try {
-    const user = await registerWithOkta(email);
+  const { email = '', _cmpConsentedState = false } = req.body;
 
+  const registrationLocation: RegistrationLocation | undefined =
+    getRegistrationLocation(req, JSON.parse(_cmpConsentedState));
```

Comment:
Just wondering why the `JSON.parse` is needed here?

Is `_cmpConsentedState` not a boolean normally?

Reply:
It was coming up as a "boolean" 
I believe formdata can only be string or blob  - https://developer.mozilla.org/en-US/docs/Web/API/FormData

Reply:
I suggest using this method then: https://github.com/guardian/gateway/blob/main/src/server/lib/queryParams.ts#L5

This converts a string to a boolean, and it's safer than using JSON.parse (as the input is coming from the client is untrusted, and JSON.parse can be unsafe)

Reply:
As something similar has caused issues with gateway in the past: https://github.com/guardian/gateway/pull/1317

Reply:
Moved isStringBoolean to it's own lib file and using instead 👍